### ISSUE TITLE
`no-whitespace-before-property` false positive

### ISSUE THREAD

**Issue by feross** [eslint/eslint#5167]
Using eslint 2.0.0-rc.0.

Code: 

```js
var arr = [1, 2, 3]
var first = arr[ 0 ]
```

Rules:

```js
{
  "rules": {
    "no-whitespace-before-property": 2
  }
}
```

Gives the following error:

```
  2:13  error  Unexpected whitespace before property 'undefined'  no-whitespace-before-property
```

The rule shouldn't consider this to be invalid whitespace, since the whitespace occurs *within* the square brackets. Spacing within square brackets is already covered by another rule.

This should be continue to be considered an error:

```js
foo [bar]
```

But not this:

```js
foo[ bar ]
```

**Comment by eslintbot** [comment]
@feross Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by ilyavolodin** [comment]
Agree. Looks like a bug. And the error message shouldn't be saying `undefined` either.

**Comment by IanVS** [comment]
We should seriously put @feross on the QA payroll, for the number of :bug: he finds.

**Comment by feross** [comment]
@IanVS Just happy to help! :+1:

This is one of the advantages of the `standard` test suite running against [real user code](https://github.com/feross/standard/blob/master/test/clone.js#L3-L9). Currently we test the top ~400 packages that use `standard`, ranked based on the # of npm dependents. You find lots of odd code out there in the wild!

It's a mixed blessing though, as your tests will fail whenever there are lint errors *in other people's code!* So the [list of repos to test](https://github.com/feross/standard-packages/blob/master/test.json) requires a bit of curation. :)

**Comment by kaicataldo** [comment]
I can look into fixing this - pretty sure I wrote that rule 😬
